### ISSUE TITLE
port FullMultijob to Github actions

### ISSUE THREAD

**Issue by poshul** [OpenMS/OpenMS#6995]

**Comment by jpfeuffer** [comment]
Just finish
https://github.com/OpenMS/OpenMS/blob/full_multijob_matrix_gha/.github/workflows/openms_ci_matrix_full.yml

**Comment by timosachsenberg** [comment]
this needs more information e.g. when should installer be built and stored as artefacts etc.

**Comment by jpfeuffer** [comment]
All in the todos

**Comment by timosachsenberg** [comment]
Thanks for drafting a big part of the full multi job.
How to go forward with the "big problem"? Artifact limits seem to be 500 MB (at most 2000 MB in total)

**Comment by jpfeuffer** [comment]
Ok this won't be enough if you plan to support another platform in the future (e.g. macOS Silicon). Maybe it's already too little because you also have to store  binaries for KNIME.
Then you probably have to create temporary folders on the archive. Or look into things like Artifactory.

**Comment by jpfeuffer** [comment]
Or you don't care if it's broken on some platforms and just upload directly.
(Won't work for KNIME which currently needs all platforms)

**Comment by timosachsenberg** [comment]
I think if we need to keep KNIME we could create temporary folders on our archive (similar as we do for our feature branch builds).

**Comment by poshul** [comment]
can we use https://github.com/lewagon/wait-on-check-action to pause uploading until all the other jobs are complete, combined with fast-fail to abort all of the runners if one fails?

**Comment by jpfeuffer** [comment]
Looks a bit complicated/flaky and wasteful on GitHub resources (every matrix branch will now take as much time as the longest). 

@timosachsenberg Are you sure that there is a limit for open source projects? https://github.com/orgs/community/discussions/26438

**Comment by timosachsenberg** [comment]
No and I guess you are right. So maybe we can just try with the artefacts (and hope they don't change the billing too soon ;)

**Comment by jpfeuffer** [comment]
One thing missing: You must still talk to IT about organizing the SSH/SCP/RSYNC (or others) to the archive. Not sure what would be the most secure way.
Note that anyone could make a PR and introduce a "rm -rf" in the ssh command. Members with write access even direct push. You either need to set strict branch protection, member access, PR approval (which I think is currently not so bad) and maybe limit the things one can do with the credentials/keys on the archive.

**Comment by timosachsenberg** [comment]
That is a good point. I think this can probably be handled by having a dedicated user on our side.

**Comment by jpfeuffer** [comment]
Yes, however, it is not easy to allow the creation of new files without allowing deletion.

**Comment by jpfeuffer** [comment]
Note that this is almost finished on the mentioned branch. Needs testing and adjusting

**Comment by jpfeuffer** [comment]
Maybe merge into develop with the PR and push triggers commented out. Then test with workflow_dispatch

**Comment by poshul** [comment]
> One thing missing: You must still talk to IT about organizing the SSH/SCP/RSYNC (or others) to the archive. Not sure what would be the most secure way. Note that anyone could make a PR and introduce a "rm -rf" in the ssh command. Members with write access even direct push. You either need to set strict branch protection, member access, PR approval (which I think is currently not so bad) and maybe limit the things one can do with the credentials/keys on the archive.

The "proper" way to do this is with setting up a key that's limited to use a specific command and rrsync. I'm gonna see if I can get that working before merging.

**Comment by poshul** [comment]
Merged so that we can run the action manually. Still needs testing and some hacking on the package building.

**Comment by timosachsenberg** [comment]
@poshul is this done? can you give an update what is missing if anything?

**Comment by poshul** [comment]
Still needs testing with the `release/*` branches. I will do this as part of the prep for 3.2
